ext/GeometryOpsProjExt/reproject.jl

```diff
@@ -22,29 +22,51 @@ function reproject(geom;
         reproject(geom, transform; kw...)
     end
 end
+struct ApplyToPoint2{F}
+    f::F
+end
+struct ApplyToPoint3{F}
+    f::F
+end
+
+(t::ApplyToPoint2)(p) = t.f(GI.x(p), GI.y(p))
+(t::ApplyToPoint3)(p) = t.f(GI.x(p), GI.y(p), GI.z(p))
+
 function reproject(geom, source_crs, target_crs;
     time=Inf,
+    threaded=False(),
     always_xy=true,
     transform=nothing,
     kw...
 )
-    transform = if isnothing(transform) 
-        s = source_crs isa Proj.CRS ? source_crs : convert(String, source_crs)
-        t = target_crs isa Proj.CRS ? target_crs : convert(String, target_crs)
-        Proj.Transformation(s, t; always_xy)
+    if istrue(threaded)
```

Comment:
I don't remember this function :D 

```suggestion
    if booltype(threaded) isa True

Reply:
Oh I have it in Rasters I think! Should add it here it's cleaner

Reply:
Yeah, that would be another PR though

Reply:
It's 4 lines ;)

Reply:
Well then we would go through and refactor every use of `boolastype isa True` which would take a while